lib/protoparser/prometheus/parser.go

```diff
@@ -147,16 +147,20 @@ func (r *Row) unmarshal(s string, tagsPool []Tag, noEscapes bool) ([]Tag, error)
 	return tagsPool, nil
 }
 
+var rowsReadScrape = metrics.NewCounter(`vm_protoparser_rows_read_total{type="promscrape"}`)
+
 func unmarshalRows(dst []Row, s string, tagsPool []Tag, noEscapes bool, errLogger func(s string)) ([]Row, []Tag) {
 	for len(s) > 0 {
 		n := strings.IndexByte(s, '\n')
 		if n < 0 {
 			// The last line.
-			return unmarshalRow(dst, s, tagsPool, noEscapes, errLogger)
+			dst, tagsPool = unmarshalRow(dst, s, tagsPool, noEscapes, errLogger)
```

Comment:
It looks like `s=""` is missing after this line. Otherwise this change may result in infinite loop.